### PR TITLE
CardComponent: Create food item variation of card component

### PR DESCRIPTION
**Pull request by carizzuti**
Add the food item variation of the card component.

image

### CODE REVIEW

**Comment by aahenry51** [comment]
Is this WIP or done?

**Comment by carizzuti** [comment]
Updated styles, added scrollview and split Card component into different components.

Screenshot_1650906134

File: mobile/src/components/card/card.type.ts [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+  subtitle?: string;
+  tags?: string[];
+  rating?: number;
+  usersVoted?: number;
+  favorite?: boolean;
+  price?: number;
+  freeDelivery?: string;
+  deliveryTime?: number;
+  restaurant?: Restaurant;
+  foodItem?: FoodItem;
```
Comment (lasharela) [inline mobile/src/components/card/card.type.ts:16]: try to avoid names related directly to context. like restaurant and foodItem.

**Review by lasharela — APPROVED** [review]

**Review by Nap1400 — APPROVED** [review]

File: mobile/src/types/data.ts [hunk 2]
```diff
@@ -1,12 +1,23 @@
 export interface Restaurant {
```
Comment (lasharela) [inline mobile/src/types/data.ts:1]: Let's use only types. 
Kind a let's keep consistency 

And in front you have to add T or I

File: mobile/src/types/data.ts [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
+  usersVoted?: number;
+  favorite?: boolean;
+  freeDelivery?: string;
+  deliveryTime?: number;
   foodItems?: string;
   imageName?: string;
   items?: string[];
 }
+
+export interface FoodItem {
```
Comment (lasharela) [inline mobile/src/types/data.ts:14]: Same

**Review by lasharela — APPROVED** [review]

**Review by Dulaya — APPROVED** [review]
I tested them, and they look good on my screen. Good job. ✔️

**Review by Josetf2396 — APPROVED** [review]

**Pull request merged**